pubsub/cli.rb

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud"
+
+# Shows using gcloud to interact with Pub/Sub
+class PubSubSample
+  attr_accessor :topic, :sub
+
+  def initialize project
+    @project = project
+    @topic = "ruby-sample"
+    @sub = "ruby-sample-sub"
+    @subp = "ruby-sample-sub-p"
+
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+
+    topic = pubsub.topic @topic, autocreate: true
+
+    sub = topic.get_subscription @sub
+    topic.subscribe @sub if sub.nil?
+
+    subp = topic.get_subscription @subp
+    topic.subscribe(
+      @subp,
+      endpoint: "https://#{@project}.appspot.com/push") if subp.nil?
+  end
+
+  def publish message
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+    topic = pubsub.topic @topic
+    topic.publish message
+  end
+
+  def pull
```

Comment:
Alternatively, the [existing Pub/Sub code samples](https://cloud.google.com/pubsub/subscriber#create) use string literals for the topic name and subscription name, etc.

One benefit of this is that end-users can see what a topic name or subscription name might look like.  eg. "Are dashes OK to use?  What might I name these in a hello world application?"

```javascript
function subscribe(callback) {
  var topicName = 'messageCenter';
  var subscriptionName = 'newMessages';

  var options = {
    reuseExisting: true
  };
  pubsub.subscribe(topicName, subscriptionName, options, callback);
}
```

I'd like to see either literals OR method parameters.  The problem with literals is that we can't use it for Project ID because our tests need to be able to override Project ID to use our test project.  So I like method params.  Thoughts?

A third alternative that might bring in the both of both worlds would be to use parameter defaults to show examples:

```ruby
def pull project = "myproject", subscription_name = "mysubscription"
  gcloud       = Gcloud.new project
  pubsub       = gcloud.pubsub
  subscription = pubsub.subscription subscription_name

  subscription.pull.each do |message|
    puts message.data
  end
end
```